internal/appsec/waf_test.go

```diff
@@ -72,6 +75,9 @@ func TestCustomRules(t *testing.T) {
 			mt := mocktracer.Start()
 			defer mt.Stop()
 
+			telemetryClient := new(telemetrytest.RecordClient)
+			defer telemetry.SwapClient(telemetry.SwapClient(telemetryClient))
```

Comment:
This looks weird locally... I had to re-read this twice to see the second `telemetry.SwapClient`... 

Optional, but

```suggestion
			telemetryClient := new(telemetrytest.RecordClient)
			prevClient := telemetry.SwapClient(telemetryClient)
			defer telemetry.SwapClient(prevClient)
```